src/NServiceBus.Core/Routing/RoutingFeature.cs

```diff
@@ -7,50 +7,50 @@
 
     class RoutingFeature : Feature
     {
-        public const string EnforceBestPracticesSettingsKey = "NServiceBus.Routing.EnforceBestPractices";
-
         public RoutingFeature()
         {
             EnableByDefault();
             Defaults(s =>
             {
-                s.SetDefault<UnicastRoutingTable>(new UnicastRoutingTable());
                 s.SetDefault<ConfiguredUnicastRoutes>(new ConfiguredUnicastRoutes());
-
-                s.SetDefault<EndpointInstances>(new EndpointInstances());
-                s.SetDefault<DistributionPolicy>(new DistributionPolicy());
-
                 s.SetDefault(EnforceBestPracticesSettingsKey, true);
-
-                s.SetDefault<Publishers>(new Publishers()); // required to initialize MessageEndpointMappings.
+                s.Set<RoutingFeature>(this);
```

Comment:
If we follow this approach should a feature auto-register itself by default in the settings?

Another idea: Since we cannot assume each feature will have the postfix feature would it make sense to "abstract" access to in a dedicated "holder"? I think it is clear as long as you do:

`context.Settings.Get<SomeFeature>().Dependency`

but assuming the feature is named `Some` then it is a bit unclear what some is when you see it accessed like

`context.Settings.Get<Some>().Dependency`

So registration would happen automatically and access to feature will not be 

`context.Settings.Get<Specific>().Dependency`

but 
`
context.Feature<Specific>().Dependency` ?